packages/wallet/ui/src/util/WalletBackendAdapter.js

```diff
@@ -130,15 +130,82 @@ export const makeWalletBridgeFromFollower = (
     ]),
   );
 
+  // We assume just one cosmos purse per brand.
+  const offers = {};
+  const brandToPurse = new Map();
+  const pursePetnameToBrand = new Map();
+
+  const updatePurses = () => {
+    const purses = [];
+    for (const [brand, purse] of brandToPurse.entries()) {
+      if (purse.currentAmount && purse.brandPetname) {
+        console.log(purse.currentAmount);
+        pursePetnameToBrand.set(purse.pursePetname, brand);
+        purses.push(purse);
+      }
+    }
+    // console.log(purses);
+    notifierKits.purses.updater.updateState(harden(purses));
+  };
+
   const followLatest = async () => {
-    for await (const { value: state } of iterateLatest(follower)) {
+    /** @type {number} */
+    let firstHeight;
+    for await (const { blockHeight } of iterateReverse(follower)) {
+      // TODO: Only set firstHeight and break if the value contains all our state.
+      firstHeight = blockHeight;
+    }
+    for await (const { value } of iterateEach(follower, {
```

Comment:
We probably don't want to update the UI until we "catch up" to the current block, but I'm not sure how we can determine that. Otherwise, we might see for instance the purse balances cycle through several different amounts while we iterate towards the latest block.

Reply:
I see your point, but don't see how to solve it easily right now.